src/utils.js

```diff
@@ -1,37 +1,51 @@
 export const geocodeByAddress = (address, callback) => {
-  const geocoder = new google.maps.Geocoder()
-  const OK = google.maps.GeocoderStatus.OK
+  return new Promise((resolve, reject) => {
+    const geocoder = new google.maps.Geocoder()
+    const OK = google.maps.GeocoderStatus.OK
 
-  geocoder.geocode({ address }, (results, status) => {
-    if (status !== OK) {
-      callback({ status }, null, results)
-      return
-    }
+    geocoder.geocode({ address }, (results, status) => {
+      if (status !== OK) {
+        if (callback) {
+          callback({ status }, null, results)
+        }
+        return reject({ error: { status }, results })
+      }
 
-    const latLng = {
-      lat: results[0].geometry.location.lat(),
-      lng: results[0].geometry.location.lng(),
-    }
+      const latLng = {
+        lat: results[0].geometry.location.lat(),
+        lng: results[0].geometry.location.lng(),
+      }
 
-    callback(null, latLng, results)
+      if (callback) {
+        callback(null, latLng, results)
+      }
+      return resolve({ latLng, results })
+    })
   })
 }
 
 export const geocodeByPlaceId = (placeId, callback) => {
```

Comment:
Same as `geocodeByAddress`. 
1. Please add warning message for callbacks.
2. reject should look like

```js
reject({ status }, results)
```

3. resolve should look like 

```js
resolve(latLng, results)
```